nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -390,17 +391,19 @@ void VoxelLayer::raytraceFreespace(
 
     double point_x, point_y, point_z;
     if (worldToMap3DFloat(wpx, wpy, wpz, point_x, point_y, point_z)) {
-      unsigned int cell_raytrace_range = cellDistance(clearing_observation.raytrace_range_);
+      unsigned int cell_raytrace_max_range = cellDistance(clearing_observation.raytrace_max_range_);
+      unsigned int cell_raytrace_min_range = cellDistance(clearing_observation.raytrace_min_range_);
+
 
       // voxel_grid_.markVoxelLine(sensor_x, sensor_y, sensor_z, point_x, point_y, point_z);
       voxel_grid_.clearVoxelLineInMap(
         sensor_x, sensor_y, sensor_z, point_x, point_y, point_z,
         costmap_,
         unknown_threshold_, mark_threshold_, FREE_SPACE, NO_INFORMATION,
-        cell_raytrace_range);
+        cell_raytrace_max_range, cell_raytrace_min_range);
 
       updateRaytraceBounds(
```

Comment:
Doesn't this need min range too to know whether it should touch the cells if its lower than the min distance?

Reply:
both here and in the 2D layer

Reply:
From what I understand, updateRaytraceBounds function updates the maximum and minimum bounds of the maps. So in that case I think only the max range would be relevant for the map bounds since it is a line. Is my train of thought incorrect?

Reply:
From what I understand, updateRaytraceBounds function updates the maximum and minimum bounds of the maps. So in that case I think only the max range would be relevant for the map bounds since it is a line. Is my train of thought incorrect?

Reply:
If in the lines above you are clearing a line, but the line is too short so you skip, I don't think that should affect the bounds of the map for update since that measurement was skipped.

Which actually reminds me that this logic is all good for clearing (e.g. if too short, don't clear), but you don't use the min range parameter in the marking side of things to skip marking if too close as well. It basically just needs something like this https://github.com/ros-planning/navigation2/blob/f86e44cedee5b6851e1aa09b20b09554d205b64a/nav2_costmap_2d/plugins/obstacle_layer.cpp#L390-L393 to check if too close (and `sq_dist` is already computed so no additional computations required)

Reply:
I mean that updating bounds with both max_range and min_range would be redundant since any updates with the max_range would override updates with the min_range

Reply:
Ah, I see the miscommunication. I don't mean update the bounds with min range, I'm saying we should skip updating bounds entirely if the `min range > point range`, so that we should entirely ignore it since it fell short.